src/NuGet.Services.AzureSearch/VersionList/FilteredVersionList.cs

```diff
@@ -154,6 +180,39 @@ private static SearchIndexChangeType UpsertToSearchIndex(Context ctx)
             return SearchIndexChangeType.UpdateVersionList;
         }
 
+        private static IReadOnlyList<HijackIndexChange> UpsertToHijackIndex(Context ctx)
+        {
+            var changes = new List<HijackIndexChange>();
+
+            // Update the metadata for the upserted version.
+            changes.Add(HijackIndexChange.UpdateMetadata(ctx.ChangedVersion));
+
+            // If the new latest is not the version that we are processing right now, explicitly set the current version
+            // to not be the latest. This supports the reflow scenario.
+            if (ctx.NewLatest != ctx.ChangedVersion)
+            {
+                changes.Add(HijackIndexChange.SetLatestToFalse(ctx.ChangedVersion));
+            }
+
+            // If the latest version has changed and the old latest version existed, mark that old latest version as
+            // no longer latest.
+            if (ctx.OldLatest != null
+                && ctx.OldLatest != ctx.NewLatest
+                && ctx.OldLatest != ctx.ChangedVersion)
+            {
+                changes.Add(HijackIndexChange.SetLatestToFalse(ctx.OldLatest));
+            }
+
+            // Always mark the new latest version as latest, even if it has not changed. This supports the reflow
+            // scenario.
+            if (ctx.NewLatest != null)
+            {
+                changes.Add(HijackIndexChange.SetLatestToTrue(ctx.NewLatest));
+            }
```

Comment:
Nope. Since `UpdateMetadata` does not imply `SetLatestToTrue` we need this for the case where the upserted version is the new latest (e.g. 1.0.0 is available, we are upserting 2.0.0).